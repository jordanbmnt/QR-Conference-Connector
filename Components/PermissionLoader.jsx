import { View } from "react-native";
import { StyleSheet } from "react-native";
import { Spinner } from "@gluestack-ui/themed";

export default PermissionsLoader = () => {
  return (
    <View style={styles.permissionLoader}>
      <Spinner size={42} />
    </View>
  );
};

const styles = StyleSheet.create({
  permissionLoader: { flex: 1, alignItems: "center", justifyContent: "center" },
});
