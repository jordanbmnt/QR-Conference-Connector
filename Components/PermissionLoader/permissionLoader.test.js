import PermissionLoader from "./PermissionLoader";
import renderer from "react-test-renderer";

describe("PermissionLoader", function () {
  const loader = renderer.create(<PermissionLoader />).toJSON();
  it("should contain a view element", function () {
    expect(loader.type).toBe("View");
  });
  it("should contain a View with one child", function () {
    expect(loader.children.length).toBe(1);
  });
  it("should contain a loader of size 42", function () {
    expect(loader.children[0].props.size).toBe(42);
  });
  it("should render the PermissionLoader component correctly", () => {
    expect(loader).toMatchSnapshot();
  });
});
