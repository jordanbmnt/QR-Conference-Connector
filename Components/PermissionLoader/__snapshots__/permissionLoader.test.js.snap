// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PermissionLoader should render the PermissionLoader component correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    aria-label="loading"
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    size={42}
    style={
      [
        undefined,
        {},
        undefined,
      ]
    }
    tabIndex={0}
  />
</View>
`;
